vmdb/app/models/host.rb

```diff
@@ -1459,10 +1459,21 @@ def refresh_patches(ssu)
   def refresh_services(ssu)
     begin
       xml = MiqXml.createDoc(:miq).root.add_element(:services)
-      services = ssu.shell_exec("chkconfig --list")
-      services = MiqLinux::Utils.parse_chkconfig_list(services)
+
+      services = ssu.shell_exec("systemctl -a --type service")
+      if services
+        # If there is a systemd use only that, chconfig is calling systemd on the background, but has misleading results
+        services = MiqLinux::Utils.parse_systemctl_list(services)
+      else
+        services = ssu.shell_exec("chkconfig --list")
+        services = MiqLinux::Utils.parse_chkconfig_list(services)
+      end
+
       services.each do |service|
-        s = xml.add_element(:service, {'name' => service[:name]})
+        s = xml.add_element(:service, {'name' => service[:name], 'systemd_load' => service[:systemd_load],
+                                       'systemd_sub' => service[:systemd_sub], 'description' => service[:description],
+                                       'running' => service[:running], 'systemd_active' => service[:systemd_active],
+                                       'typename' => service[:typename]})
```

Comment:
Also here, one per line with alignment.  Also, you don't need the curly braces around the Hash (but it doens't hurt either)